### ISSUE TITLE
Container Push with Container Load image fails.

### ISSUE THREAD

**Issue by afking** [bazelbuild/rules_docker#298]
Given my WORKSPACE, where image.tar is the `docker save` from the busybox image:

```python
container_pull(
    name = "busybox",
    registry = "index.docker.io",
    repository = "library/busybox",
    tag = "latest",
) 

container_load(
    name = "busybox_test",
    file = "//:image.tar",
)
```

And then a BUILD file:

```python
container_bundle(
    name = "bundle_test",
    images = {
        "ACCOUNT.REGION.amazonaws.com/repo:busybox": "@busybox//image",
        "ACCOUNT.REGION.amazonaws.com/repo:busybox_test": "@busybox_test//image",
    },
)

container_push(
    name = "export_test",
    bundle = ":bundle_test",
    format = "Docker",
)
```

Bazel fails with:

```sh
bazel run :export_test
INFO: Analysed target //docker:export_test (23 packages loaded).
INFO: Found 1 target...
Target //docker:export_test up-to-date:
  bazel-bin/docker/export_test
INFO: Elapsed time: 7.306s, Critical Path: 0.21s
INFO: Build completed successfully, 10 total actions

INFO: Running command line: bazel-bin/docker/export_test
ACCOUNT.REGION.amazonaws.com/repo:busybox was published with digest: sha256:f353bdf4866c5e5b5c76a97e61312cab1a788b58ce3ab7a2f31a87844a0f4093
E0129 17:08:25.815792   10712 docker_session_.py:335] Error during upload of: ACCOUNT.REGION.amazonaws.com/repo:busybox_test
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/com_jazznetworks_repo_jazz_main/../containerregistry/tools/fast_pusher_.py", line 149, in <module>
    main()
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/com_jazznetworks_repo_jazz_main/../containerregistry/tools/fast_pusher_.py", line 140, in main
    session.upload(v2_2_img)
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/containerregistry/client/v2_2/docker_session_.py", line 324, in upload
    future.result()
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/concurrent/futures/_base.py", line 398, in result
    return self.__get_result()
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/containerregistry/client/v2_2/docker_session_.py", line 285, in _upload_one
    self._put_blob(image, digest)
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/containerregistry/client/v2_2/docker_session_.py", line 211, in _put_blob
    self._patch_upload(image, digest)
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/containerregistry/client/v2_2/docker_session_.py", line 182, in _patch_upload
    accepted_codes=[httplib.CREATED])
  File "/private/var/tmp/_bazel_edwardmcfarlane/ad32b996cb424fa4cc44a5c86220cb2f/execroot/com_jazznetworks_repo_jazz_main/bazel-out/darwin-fastbuild/bin/docker/export_test.runfiles/containerregistry/client/v2_2/docker_http_.py", line 376, in Request
    raise V2DiagnosticException(resp, content)
containerregistry.client.v2_2.docker_http_.V2DiagnosticException: response: {'status': '400', 'content-length': '125', 'connection': 'keep-alive', 'docker-distribution-api-version': 'registry/2.0', 'date': 'Mon, 29 Jan 2018 17:08:25 GMT', 'content-type': 'application/json; charset=utf-8'}
None of the provided layer digests match the calculated layer digest: None
ERROR: Non-zero return code '1' from command: Process exited with status
```

Running the bundle exports the images fine.

```sh
 bazel run //docker:bundle_test
INFO: Analysed target //docker:bundle_test (1 packages loaded).
INFO: Found 1 target...
Target //docker:bundle_test up-to-date (nothing to build)
INFO: Elapsed time: 0.723s, Critical Path: 0.23s
INFO: Build completed successfully, 3 total actions

INFO: Running command line: bazel-bin/docker/bundle_test
Loaded image ID: sha256:6ad733544a6317992a6fac4eb19fe1df577d4dec7529efec28a5bd0edad0fd30
Loaded image ID: sha256:6ad733544a6317992a6fac4eb19fe1df577d4dec7529efec28a5bd0edad0fd30
Tagging 6ad733544a6317992a6fac4eb19fe1df577d4dec7529efec28a5bd0edad0fd30 as ACCOUNT.REGION.amazonaws.com/repo:busybox
Tagging 6ad733544a6317992a6fac4eb19fe1df577d4dec7529efec28a5bd0edad0fd30 as ACCOUNT.REGION.amazonaws.com/repo:busybox_test
```
Tested with AWS & Gitlab registries.

**Comment by afking** [comment]
Forgot I was using the `push_all.bzl` rule but seems to be the same with the single `container_push` rule.

**Comment by afking** [comment]
And it works with master and the `docker.docker.bzl` load! :D Thanks for fixing.

**Comment by afking** [comment]
Reopening, error fails silently:
```sh
ACCOUNT.REGION.amazonaws.com/repo:busybox was published with digest: sha256:f353bdf4866c5e5b5c76a97e61312cab1a788b58ce3ab7a2f31a87844a0f4093
E0129 17:36:21.454334   11432 docker_session_.py:335] Error during upload of: ACCOUNT.REGION.amazonaws.com/repo:busybox_test
F0129 17:36:21.455245   11432 fast_pusher_.py:156] Error publishing ACCOUNT.REGION.amazonaws.com/repo:busybox_test: response: {'status': '400', 'content-length': '125', 'connection': 'keep-alive', 'docker-distribution-api-version': 'registry/2.0', 'date': 'Mon, 29 Jan 2018 17:36:21 GMT', 'content-type': 'application/json; charset=utf-8'}
None of the provided layer digests match the calculated layer digest: None
```

**Comment by chrismcbride** [comment]
I would guess this is related to this issue: https://github.com/bazelbuild/rules_docker/issues/450

**Comment by nlopezgi** [comment]
Is this now fixed with #450? Let me know if not (with update about issue) to reopen.
